iac-extensions/terraform/src/test/resources/checks/DisabledMfaBucketDeletionCheck/aws_provider_v4.tf

```diff
@@ -0,0 +1,52 @@
+resource "aws_s3_bucket" "no_versioning" {
+  bucket = "mycompliantbuckets6255"
+}
+
+resource "aws_s3_bucket" "mynoncompliantbuckets6255" {
+  bucket = "mynoncompliantbuckets6255"
+
+  versioning {}
+}
+
+resource "aws_s3_bucket" "mycompliantbuckets6255" {
+  #      ^^^^^^^^^^^^^^^> {{Related bucket}}
+  bucket = "mycompliantbuckets6255"
+
+  versioning {
+    mfa_delete = false # Noncompliant {{Make sure allowing object deletion without MFA is safe here.}}
+#   ^^^^^^^^^^^^^^^^^^
+  }
+}
+
+resource "aws_s3_bucket" "mycompliantbuckets6255" {
+  bucket = "mycompliantbuckets6255"
+
+  versioning {
+    enabled = true
+    mfa_delete = true
+  }
+}
+
+resource "aws_s3_bucket_versioning" "non_compliant_versioning_disabled" {
+  versioning_configuration {
+    mfa_delete = "Disabled" # Noncompliant {{Make sure allowing object deletion without MFA is safe here.}}
+#   ^^^^^^^^^^^^^^^^^^^^^^^
+  }
+}
+
+resource "aws_s3_bucket_versioning" "non_compliant_versioning_missing" {
+  versioning_configuration { # Noncompliant {{Make sure allowing object deletion without MFA is safe here.}}
+# ^^^^^^^^^^^^^^^^^^^^^^^^
+  }
+}
+
+resource "aws_s3_bucket_versioning" "compliant_versioning" {
+  versioning_configuration {
+    mfa_delete = "Enabled"
+  }
+}
+
+
+resource "not_a_bucket" "name" {
+  bucket = "mycompliantbuckets6255"
+}
```

Comment:
I don't see the value of this test. 
I think it could be useful if the only difference with a noncompliant case was the resource type.